File: src/main/java/de/fraunhofer/aisec/codyze/analysis/TypestateMode.java

package de.fraunhofer.aisec.codyze.analysis;

public enum TypestateMode {

	/**
	 * Non-deterministic finite automaton. Intraprocedural, not alias-aware.
	 */
	NFA,

	/**
	 * Weighted Pushdown System. Interprocedural, alias-aware, context-aware.
	 */
	WPDS

}

File: codyze_v3/codyze-core/src/main/kotlin/config/enums/TypestateMode.kt
package de.fraunhofer.aisec.codyze_core.config.enums

/** Enum containing all supported typestate modes. */
enum class TypestateMode {
    /** Non-deterministic finite automaton. Intraprocedural, not alias-aware. */
    DFA,

    /** Weighted Pushdown System. Interprocedural, alias-aware, context-aware. */
    WPDS
}
